Lib9c/Action/HackAndSlashSweep.cs

```diff
@@ -0,0 +1,193 @@
+            var inventoryAddress = avatarAddress.Derive(LegacyInventoryKey);
+            var questListAddress = avatarAddress.Derive(LegacyQuestListKey);
+            if (context.Rehearsal)
+            {
+                return states
+                    .SetState(inventoryAddress, MarkChanged)
+                    .SetState(questListAddress, MarkChanged)
+                    .SetState(avatarAddress, MarkChanged)
+                    .SetState(context.Signer, MarkChanged);
+            }
+
+            var addressesHex = GetSignerAndOtherAddressesHex(context, avatarAddress);
+
+            if (apStoneCount > UsableApStoneCount)
+            {
+                throw new UsageLimitExceedException($"Exceeded the amount of ap stones that can be used " +
+                                                    $"apStoneCount : {apStoneCount} > UsableApStoneCount : {UsableApStoneCount}");
+            }
+
+            if (worldId == GameConfig.MimisbrunnrWorldId)
+            {
+                throw new InvalidWorldException(
+                    $"{addressesHex} [{worldId}] can't execute HackAndSlashSweep action.");
+            }
+
+            if (!states.TryGetAvatarStateV2(context.Signer, avatarAddress, out var avatarState, out var migrationRequired))
+            {
+                throw new FailedLoadStateException($"{addressesHex}Aborted as the avatar state of the signer was failed to load.");
+            }
+
+            var sheets = states.GetSheets(
+                containQuestSheet: false,
+                containStageSimulatorSheets: false,
+                sheetTypes: new[]
+                {
+                    typeof(WorldSheet),
+                    typeof(StageSheet),
+                    typeof(MaterialItemSheet),
+                    typeof(StageWaveSheet),
+                    typeof(CharacterLevelSheet),
+                });
+
+            var worldSheet = sheets.GetSheet<WorldSheet>();
+            if (!worldSheet.TryGetValue(worldId, out var worldRow, false))
+            {
+                throw new SheetRowNotFoundException(addressesHex, nameof(WorldSheet), worldId);
+            }
+
+            if (stageId < worldRow.StageBegin ||
+                stageId > worldRow.StageEnd)
+            {
+                throw new SheetRowColumnException(
+                    $"{addressesHex}{worldId} world is not contains {worldRow.Id} stage: " +
+                    $"{worldRow.StageBegin}-{worldRow.StageEnd}");
+            }
+
+            if (!sheets.GetSheet<StageSheet>().TryGetValue(stageId, out var stageRow))
+            {
+                throw new SheetRowNotFoundException(addressesHex, nameof(StageSheet), stageId);
+            }
+
+            var worldInformation = avatarState.worldInformation;
+            if (!worldInformation.TryGetWorld(worldId, out var world))
+            {
+                worldInformation.AddAndUnlockNewWorld(worldRow, context.BlockIndex, worldSheet);
+            }
```

Comment:
I think we need to add setState for worldInformation or delete this part.

Reply:
I will replace this line ` worldInformation.AddAndUnlockNewWorld(worldRow, context.BlockIndex, worldSheet);` to throw error

Reply:
fixed it. please review again 🙏🏻 @ipdae